src/issuance.rs

```diff
@@ -653,6 +671,54 @@ mod tests {
         )
     }
 
+    fn generate_identity_point_asset_base() -> AssetBase {
+        let identity_point =
+            (pallas::Point::generator() * -pallas::Scalar::one()) + pallas::Point::generator();
+        AssetBase::from_bytes(&identity_point.to_bytes()).unwrap()
+    }
+
+    fn identity_point_asset_base_test_params(
+        note1_value: u64,
+        note2_value: u64,
+        finalize: bool,
+    ) -> (
+        OsRng,
+        IssuanceAuthorizingKey,
+        IssuanceValidatingKey,
+        IssueAction,
+        IssueBundle<Unauthorized>,
+        [u8; 32],
+    ) {
+        let (mut rng, isk, ik, recipient, sighash) = setup_params();
+
+        let asset = generate_identity_point_asset_base();
+
+        let note1 = Note::new(
+            recipient,
+            NoteValue::from_raw(note1_value),
+            asset,
+            Nullifier::dummy(&mut rng),
+            &mut rng,
+        );
+
+        let note2 = Note::new(
+            recipient,
+            NoteValue::from_raw(note2_value),
+            asset,
+            Nullifier::dummy(&mut rng),
+            &mut rng,
+        );
+
+        let action =
+            IssueAction::from_parts("arbitrary asset_desc".into(), vec![note1, note2], finalize);
+
+        let ik2 = ik.clone();
+        let action2 = action.clone();
+        let bundle = IssueBundle::from_parts(ik2, NonEmpty::new(action2), Unauthorized);
+
+        (rng, isk, ik, action, bundle, sighash)
```

Comment:
remove `ik` and `action` and the redundent cloning. When needed extract them from the bundle
```rust
let (_, _, _, _, b, _) = identity_point_asset_base_test_params(10, 20, false);
        
        assert_eq!(
            b.actions.head.verify_supply(&b.ik()),
            Err(AssetBaseCannotBeIdentityPoint)
        );

```

Reply:
Done.